### PR TITLE
Prepare OS X support

### PR DESCRIPTION
**Pull request by MrAnno**
Fixes: #805
Fixes: #808 
Partial fix for #821
Related to #937

### CODE REVIEW

**Comment by ihrwein** [comment]
:+1:

**Comment by lbudai** [comment]
hi,

please wait a little bit with the merge.
I'll have a question about the way how we are getting the time .

**Comment by bkil-syslogng** [comment]
:+1: other than what @lbudai asked

**Comment by MrAnno** [comment]
I moved the `get_monotonic_time` function to the `compat` lib. Sorry for the early rebase, I didn't think it through.

File: lib/compat/time.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  clock_serv_t clock_server;
+  mach_timespec_t mach_timestamp;
+  host_get_clock_service(mach_host_self(), SYSTEM_CLOCK, &clock_server);
+  clock_get_time(clock_server, &mach_timestamp);
+  mach_port_deallocate(mach_task_self(), clock_server);
+  timestamp.tv_sec = mach_timestamp.tv_sec;
+  timestamp.tv_nsec = mach_timestamp.tv_nsec;
+#else
+  clock_gettime(CLOCK_MONOTONIC, &timestamp);
+#endif
```
Comment (bazsi) [inline lib/compat/time.c]: If you intend to move this function into combat then please use and adhere to the POSIX prototype (e.g. implement clock_gettime() and not a wrapper). This is the convention we use in other parts of contrib.

If you want to create a wrapper then do that in lib and not compat.

Thanms

**Comment by lbudai** [comment]
@MrAnno : could you squash the last patch?

**Comment by MrAnno** [comment]
squashed, rebased to master

**Pull request merged**